### ISSUE TITLE
Code not running

### ISSUE THREAD

**Issue by rtnlsltn** [0str1ch/blackhole#27]
Hey, I'm sure this project is super dead to you, but I'd love to get it working for something I'm working on. I haven't been able to get anything more than a black screen and a bunch of errors after dependency update. I'm able to use the model in my three js project, but wanted to see how all your additions (and animations) work out together.

Thanks!

**Comment by 0str1ch** [comment]
I attempted to do the same about a year ago and ran into the same issues. I opted to start fresh instead of untangling the issue that I think stems from breaking changes in R3F over the years as this project sat dormant.

I'm currently refactoring this entire project based on DNGR (Double Negative Gravitational Renderer), the renderer created for the movie Interstellar and its supermassive black hole. To visualize a blackhole with DNGR required several hundred servers at a time in a render farm 10+ years ago, so I've been attempting to dumb it down for real-time performance while retaining the core principles. More info in the paper here: [https://arxiv.org/abs/1502.03808](https://arxiv.org/abs/1502.03808)
